### PR TITLE
Detect duplicate versions

### PR DESCRIPTION
**Pull request by josh**
Adding some CI enforcement around packages installing multiple versions of conflicting dependencies.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -8,3 +8,16 @@ for path in examples/*; do
...
+
+ensure_dedupe() {
+  if [ $(npm list --parseable "$1" | wc -l) -ne 1 ]; then
+    echo "$1: duplicate versions detected"
+    npm list "$1"
+    exit 1
+  fi
+}
+
+ensure_dedupe "graphql"
```
Comment (josh) [inline test.sh]:
```
graphql: duplicate versions detected
webpack-config-github@0.6.0 /home/travis/build/github/webpack-config-github
├─┬ babel-plugin-relay@1.5.0
│ └── graphql@0.13.1  deduped
├─┬ eslint-plugin-github@0.23.0
│ ├─┬ eslint-plugin-graphql@1.5.0
│ │ └─┬ graphql-config@1.0.9
│ │   └── graphql@0.11.7 
│ └─┬ eslint-plugin-relay@0.0.21
│   └── graphql@0.13.1  deduped
├── graphql@0.13.1 
└─┬ relay-compiler@1.5.0
  └── graphql@0.13.1  deduped
```

`eslint-plugin-graphql` is problematic with it's conflicting `graphql` package.

Comment (koddsson) [reply test.sh]: Is it just a matter of updating `graphql` in `eslint-plugin-graphql`?

Comment (josh) [reply test.sh]: > Is it just a matter of updating graphql in eslint-plugin-graphql?

Well, they haven't published a compatible version yet.

Comment (koddsson) [reply test.sh]: I was thinking maybe we should make a PR upstream 😛

Comment (josh) [reply test.sh]: Yeah, we're going to be blocked on the linter crashing with a parse error on the new schema format. They really didn't roll out this backwards incompatible schema format change in a good way.

**Pull request merged**